pkg/machinery/resources/network/network.go

```diff
@@ -36,17 +36,21 @@ func LinkID(linkName string) string {
 }
 
 // RouteID builds ID (primary key) for the route.
-func RouteID(table nethelpers.RoutingTable, family nethelpers.Family, destination netip.Prefix, gateway netip.Addr, priority uint32) string {
+func RouteID(table nethelpers.RoutingTable, family nethelpers.Family, destination netip.Prefix, gateway netip.Addr, priority uint32, outLinkName string) string {
 	dst, _ := destination.MarshalText() //nolint:errcheck
 	gw, _ := gateway.MarshalText()      //nolint:errcheck
 
-	tablePrefix := ""
+	prefix := ""
 
 	if table != nethelpers.TableMain {
-		tablePrefix = fmt.Sprintf("%s/", table)
+		prefix = fmt.Sprintf("%s/", table)
 	}
 
-	return fmt.Sprintf("%s%s/%s/%s/%d", tablePrefix, family, string(gw), string(dst), priority)
+	if family == nethelpers.FamilyInet6 {
+		prefix += fmt.Sprintf("%s/", outLinkName)
```

Comment:
why do we add the link name now, i guess it was missed before?

Reply:
IPv6 allows onlink routes to the same address on different interfaces, so Talos incorrectly was trying to put all such route into a single `RouteStatus` resource causing an avalanche of useless updates, and mis-reporting routes as well.